### PR TITLE
Clean-up, refactoring, update changes html encode.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] add WithEncodeTags.
- [x] code coverage 100%.

### CODE REVIEW

File: docs/message.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ $this->setJsFiles($assetManager->getJsFiles());
     ->headerMessage('Very important')
     ->body('<strong>Holy guacamole!</strong> You should check in on some of those fields below.')
     ->size('is-large')
-    ->options(['class' => 'has-text-justified']) ?>
+    ->withOptions(['class' => 'has-text-justified']) ?>
```
Comment (samdark) [inline docs/message.md:45]: Adding `with` to widget methods isn't a good idea. Let's revert these naming changes.

**Review by samdark — CHANGES_REQUESTED** [review]

File: docs/message.md (hunk 1 above)
Comment (terabytesoftw) [reply docs/message.md]: done.

File: docs/message.md [hunk 2]
```diff
@@ -61,16 +61,17 @@ The code above generates the following HTML:
...
 `optionsCloseButton(array $value)`| Options for rendering the close button tag. | `[]`
-`optionsHeader(array $value)`     | HTML attributes for the widget header tag. | `[]`
-`size(string $value)`             | Message widget size. Default is normal. Options available are: `is-small`, `is-medium`, `is-large`. | `normal`
-`withoutCloseButton(bool $value)` | Allows you to disable close button. | `false`
-`withoutHeader(bool $value)`      | Allows you to disable header. | `true`
+`optionsHeader(array $value)` | HTML attributes for the widget header tag. | `[]`
+`size(string $value)` | Message widget size. Default is normal. Options available are: `is-small`, `is-medium`, `is-large`.  | `normal`
+`closeButton()` | Allows you to enable close button. | `true`
+`withoutHeader()` | Allows you to disable header. | `false`
+`encodeTags()` | Allows you to enable the encoding tags html. | `true`
```
Comment (samdark) [inline docs/message.md:77]: I think the option here is useless.

File: docs/navbar.md [hunk 3]
```diff
@@ -128,17 +128,18 @@ HTML produced is like the following:
...
+`brandUrl(string $value)` | "href" of the brand's link. | `/`
+`toggleIcon(string $value)` | Toggle icon. | `<span aria-hidden='true'></span>`
+`options(array $value)` | HTML attributes for the nav tag. | [`class` => `navbar`]
+`optionsBrand(array $value)` | HTML attributes of the brand tag. | [`class` => `navbar-brand`]
+`optionsBrandLabel` | HTML attributes of the brand label tag. | [`class` => `navbar-item`]
+`optionsBrandImage` | HTML attributes of the brand link tag.           | [`class` => `navbar-item`]
+`optionsItems(array $value)` | HTML attributes of the nav items tag. | [`class` => `navbar-start`] or [`class` => `navbar-end`]
+`optionsMenu(array $value)` | HTML attributes of the nav menu tag. | [`class` => `navbar-menu`]
+`optionsToggle(array $value)`| HTML attributes of the navbar toggle button tag. | [`aria-expanded` => `false`, `aria-label` => `menu`, `class` => `navbar-burger`, `role` => `button`]
+`encodeTags()` | Allows you to enable the encoding tags html. | `true`
```
Comment (samdark) [inline docs/navbar.md:145]: Does it make sense to turn it on? That would render image as `<img ...`.

File: docs/panel.md [hunk 4]
```diff
@@ -117,13 +117,14 @@ HTML produced is like the following:
...
+-------|-------------|---------
+ `options(array $value)` | HTML attributes for the widget container tag. | [`class` => `panel`]
+ `headingOptions(array $value)` | HTML attributes of the heading. | [`class` => `panel-heading`]
+ `heading(string $value)` | Text of the brand heading. | `''`
+ `color(string $value)` | Color panel. | `is-primary`, `is-link`, `is-info`, `is-success`, `is-warning`, `is-danger`
+ `tabs(array $value)` | List of panel tabs items. | `[]`
+ `tabsOptions(array $value)` | HTML attributes for the tabs container tag. | `[]`
+ `withoutEncodeLabels()` | When tags Labels HTML should not be encoded. | `false`
+ `template(string $value)` | String the template for rendering panel. | `{panelBegin}{panelHeading}{panelTabs}{panelItems}{panelEnd}`
+ `encodeTags()` | Allows you to enable the encoding tags html. | `true`
```
Comment (samdark) [inline docs/panel.md:130]: Looks like it doesn't make sense to turn it on.

File: docs/tabs.md [hunk 5]
```diff
@@ -96,14 +96,15 @@ HTML produced is like the following:
...
+`options(array $value)` | HTML attributes for the widget container tag. | [`class` => `tabs`]
+`items(array $value)` | List of tab items. | `[]`
+`currentPath(string $value)` | Allows you to assign the current path of the URL from request controller. | ``
+`withoutActivateItems()` | Disable activate items according to whether their currentPath. | `false`
+`withoutEncodeLabels()` | When tags Labels HTML should not be encoded. | `false`       |
+`size(string $value)` | Size of the tabs list. | `is-small`, `is-medium`, `is-large`
+`alignment(string $value)` | Alignment of the tabs list. | `''`
+`style(string $value)` | Style of the tabs list. | `''`
+`tabsContentOptions(array $value)` | List of HTML attributes for the `tabs-content` container. | `[]`
+`encodeTags()` | Allows you to enable the encoding tags html. | `true`
```
Comment (samdark) [inline docs/tabs.md:110]: It makes no sense to turn it on.

**Review by samdark — CHANGES_REQUESTED** [review]

File: docs/menu.md [hunk 6]
```diff
@@ -97,22 +97,22 @@ HTML produced is like the following:
...
+-------|-------------|---------
+`id(string $value)` | Widget ID. | `''`
+`withoutActivateItems()` | Disable activate items according to whether their currentPath. | `false`
+`activateParents()` | Whether to activate parent menu items when one of the corresponding child menu items is active. | `true`
+`activeCssClass(string $value)` | The CSS class to be appended to the active menu item. | `is-active`
+`brand(string $value)` | Custom brand content. | `''`
+`currentPath(string $value)` | Allows you to assign the current path of the URL from request controller. | `''`
+`withoutEncodeLabels()` | Disable encoding for labels. | `false`
+`firstItemCssClass(string $value)` | The CSS class for the first item in the main menu or each submenu. | `''`
+`withoutHideEmptyItems()` | Whether to hide empty menu items. An empty menu item is one whose `url` option is not set and which has no visible child menu items. | `false`
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply docs/menu.md:110]: done.

File: docs/tabs.md (hunk 5 above)
Comment (terabytesoftw) [reply docs/tabs.md:110]: done.

File: docs/panel.md (hunk 4 above)
Comment (terabytesoftw) [reply docs/panel.md:130]: done.

File: docs/navbar.md (hunk 3 above)
Comment (terabytesoftw) [reply docs/navbar.md:145]: done.

File: docs/message.md (hunk 2 above)
Comment (terabytesoftw) [reply docs/message.md:77]: done.

**Comment by samdark** [comment]
👍

**Pull request merged**